packages/common/src/services/audius-api-client/ResponseAdapter.ts

```diff
@@ -390,13 +391,17 @@ export const makePlaylist = (
 }
 
 export const makeActivity = (
-  activity: APIActivity
+  activity: APIActivity | APIActivityV2
 ): UserTrackMetadata | UserCollectionMetadata | undefined => {
-  switch (activity.item_type) {
+  const itemType =
+    'item_type' in activity ? activity.item_type : activity.itemType
+  switch (itemType) {
     case 'track':
-      return makeTrack(activity.item)
+      // Cast to APITrack because TS is not correctly inferring the type
+      return makeTrack(activity.item as APITrack)
```

Comment:
interesting, just curious what .item is before the cast?

Reply:
I wonder if you could use a TS type guard to be explicit about `APIActivity` vs `APIActivityV2`, that may help with the inference here

here's an example https://github.com/AudiusProject/audius-protocol/blob/main/libs/src/sdk/types/File.ts#L31